analyses/infercnv-consensus-cell-type/exploratory-notebooks/02_ewings-reference-cnv.Rmd

```diff
@@ -0,0 +1,234 @@
+    stringr::str_split_i(basename(x), "_", 1)
+  })
+```
+
+
+### Read input files
+
+
+We'll first read in the cell type annotations:
+```{r}
+celltype_df <- celltype_files |>
+  purrr::map(readr::read_tsv) |>
+  purrr::list_rbind(names_to = "library_id") |>
+  dplyr::mutate(cell_id = glue::glue("{library_id}-{barcodes}")) |>
+  dplyr::select(cell_id, consensus_annotation, ewing_annotation)
+```
+
+
+Now, we'll read in the `inferCNV` results:
+```{r}
+infercnv_df <- infercnv_files |>
+  purrr::map(prepare_cnv_df) |>
+  purrr::list_rbind(names_to = "infercnv_library_id") |>
+  dplyr::inner_join(celltype_df, by = "cell_id") |>
+  # condense labels for cell type groupings
+  dplyr::mutate(
+    label = ifelse(
+      stringr::str_detect(ewing_annotation, "tumor"),
+      "tumor",
+      cell_group
+    ),
+    label = forcats::fct_relevel(label, "unknown", "tumor", "reference")
+  )
+```
+
+
+## Analysis
+
+We'll first visualize the total CNV distributions for each inference for each grouping of cells:
+
+- `unknown`: query cells
+- `tumor`: query cells labeled "tumor" by `cell-type-ewings`
+- `reference`: cells in the normal reference
+
+```{r, fig.height = 8}
+plot_cnv_distributions(infercnv_df, "endo_pooled")
+plot_cnv_distributions(infercnv_df, "immune_pooled")
+plot_cnv_distributions(infercnv_df, "endo-immune_pooled")
+```
+
+
+* The `endo` reference cell CNV distributions generally do not overlap with tumor cell CNV distributions.
+* The `endo-immune` and `immune` reference cell CNV distributions do have a wider spread but do not _substantially_ overlap with the tumor cell CNV distributions. 
+
+Let's look more closely at the `immune` reference to see if any particular libraries might be driving a (minor) trend.
+For each inference with the pooled `immune` reference, we'll find the 25\% top reference cells with the highest CNVs.
+Do these cells tend to be from a particular library?
+
+First, note how many reference cells come from each library:
+
+```{r}
+ref_immune_df <- infercnv_df |>
+  dplyr::filter(
+    normal_reference == "immune_pooled",
+    cell_group == "reference"
+  ) |>
+  tidyr::separate(cell_id, into = c("reference_library_id", "barcode"), sep = "-", remove = FALSE)
+
+
+ref_immune_df |>
+  dplyr::select(cell_id) |>
+  dplyr::distinct() |>
+  tidyr::separate(cell_id, into = c("reference_library_id", "barcode"), sep = "-") |>
+  dplyr::count(reference_library_id)
+```
+
+The table displayed below shows for each inference (`infercnv_library_id`), how many of top 25\% CNV cells in the pooled immune reference are from each library.
+For example, of the reference cells with the highest CNVs in the `SCPCL000822` inference, 28 cells came from `SCPCL000822`, 30 cells came from `SCPCL000824`, etc.
+Generally, these numbers are consistent across inferences although cells from `SCPCL000824` do seem to be more prevalent in certain samples.
+Even so, there doesn't seem to be any major trend here.
+
+```{r}
+top25 <- ref_immune_df |>
```

Comment:
Maybe rather than showing the total number of cells it would make sense to show the proportion of all cells from that library compared to cells that are in the top 25%? I would expect that generally this should be a small number and if we see any glaring numbers then that might be concerning. That being said, I think the signal is pretty low and we do see separation between tumor and normal for the most part.